### PR TITLE
[refactor] #4126: test to prevent cross chain replay attacks

### PR DESCRIPTION
**Pull request by timofeevmd**
### Linked issue

Closes #{issue_number} 

### Checklist

- [ ] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: client/tests/integration/mod.rs [hunk 1]
```diff
@@ -22,6 +22,7 @@ mod set_parameter;
 mod sorting;
 mod transfer_asset;
 mod triggers;
+mod tx_chainid;
```
Comment (Arjentix) [inline client/tests/integration/mod.rs:25]:
```suggestion
mod tx_chain_id;
```

File: client/tests/integration/tx_chainid.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+use test_network::*;
+
+use crate::integration::asset::asset_id_new;
+
+#[test]
+fn send_tx_with_different_chain_id() {
+    let (_rt, _peer, test_client) = <PeerBuilder>::new().with_port(11_240).start_with_runtime();
+    wait_for_genesis_committed(&[test_client.clone()], 0);
+    // Given
+    let sender_account_id = AccountId::from_str("sender@wonderland").expect("Valid");
```
Comment (Arjentix) [inline client/tests/integration/tx_chainid.rs:14]:
```suggestion
    let sender_account_id = AccountId::from_str("sender@wonderland").unwrap();
```

Please, write like so in a new code

Comment (Arjentix) [reply client/tests/integration/tx_chainid.rs:14]: Same for other places in this file

File: client/tests/integration/tx_chainid.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    let sender_keypair = KeyPair::generate().expect("Failed to generate sender KeyPair.");
+    let receiver_account_id = AccountId::from_str("receiver@wonderland").expect("Valid");
+    let asset_definition_id = AssetDefinitionId::from_str("test_asset#wonderland").expect("Valid");
+    let to_transfer = 1;
+    let create_sender_accout = Register::account(Account::new(
```
Comment (Arjentix) [inline client/tests/integration/tx_chainid.rs:19]:
```suggestion
    let create_sender_account = Register::account(Account::new(
```

File: client/tests/integration/tx_chainid.rs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        [sender_keypair.public_key().clone()],
+    ));
+    let create_receiver_account = Register::account(Account::new(receiver_account_id.clone(), []));
+    let create_asset =
+        Register::asset_definition(AssetDefinition::quantity(asset_definition_id.clone()));
+    let register_asset: InstructionBox = Register::asset(Asset::new(
+        AssetId::new(asset_definition_id.clone(), sender_account_id.clone()),
+        AssetValue::Quantity(10),
+    ))
+    .into();
```
Comment (Arjentix) [inline client/tests/integration/tx_chainid.rs:30]: Quite misleading variable names. I suggest this:

```suggestion
    let register_asset_definition =
        Register::asset_definition(AssetDefinition::quantity(asset_definition_id.clone()));
    let register_asset: InstructionBox = Register::asset(Asset::new(
        AssetId::new(asset_definition_id.clone(), sender_account_id.clone()),
        AssetValue::Quantity(10),
    ))
    .into();
```

File: client/tests/integration/tx_chainid.rs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        .expect("Failed to create sender account.");
+    test_client
+        .submit_blocking(create_receiver_account)
+        .expect("Failed to create receiver account.");
+    test_client
+        .submit_blocking(create_asset)
+        .expect("Failed to create asset");
+    test_client
+        .submit_blocking(register_asset.clone())
+        .expect("Failed to register asset");
```
Comment (Arjentix) [inline client/tests/integration/tx_chainid.rs:43]:
```suggestion
    test_client
        .submit_all_blocking([
            create_sender_accout,
            create_receiver_account,
            create_asset,
            register_asset,
        ])
        .expect("Failed to sumbit precondition instructions");
```

File: client/tests/integration/tx_chainid.rs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    let chain_id_0 = ChainId::new("0");
```
Comment (Arjentix) [inline client/tests/integration/tx_chainid.rs:44]:
```suggestion
    let chain_id_0 = ChainId::new("0"); // Value configured by default
```

Comment (Arjentix) [reply client/tests/integration/tx_chainid.rs:44]: Just to explain why it works for `0`

File: client/tests/integration/tx_chainid.rs (hunk 5 above)
Comment (timofeevmd) [reply client/tests/integration/tx_chainid.rs:43]: left in separate checks due to data type mismatch

**Pull request closed without merging**